code/modules/mob/living/basic/festivus_pole.dm

```diff
@@ -0,0 +1,110 @@
+	icon_gib = "festivus_pole"
+	health_doll_icon = "festivus_pole"
+	gender = NEUTER
+	gold_core_spawnable = HOSTILE_SPAWN
+	basic_mob_flags = DEL_ON_DEATH
+
+	response_help_continuous = "rubs"
+	response_help_simple = "rub"
+	response_disarm_continuous = "pushes"
+	response_disarm_simple = "push"
+
+	mob_size = MOB_SIZE_LARGE
+	pixel_x = -16
+	base_pixel_x = -16
+
+	speed = 1
+	maxHealth = 200
+	health = 200
+	melee_damage_lower = 8
+	melee_damage_upper = 12
+	attack_verb_continuous = "bites"
+	attack_verb_simple = "bite"
+	attack_sound = 'sound/weapons/bite.ogg'
+	attack_vis_effect = ATTACK_EFFECT_BITE
+
+	faction = list(FACTION_HOSTILE)
+	speak_emote = list("polls")
+
+	death_message = "is hacked into pieces!"
+
+	ai_controller = /datum/ai_controller/basic_controller/festivus_pole
+
+	///how much charge we give off to cells around us when rubbed
+	var/recharge_value = 75
+
+/mob/living/basic/festivus/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/death_drops, list(/obj/item/stack/rods))
+
+/datum/ai_controller/basic_controller/festivus_pole
+	blackboard = list(
+		BB_TARGETTING_DATUM = new /datum/targetting_datum/basic(),
+		BB_LOW_PRIORITY_HUNTING_TARGET = null, // APCs
+	)
+
+	ai_movement = /datum/ai_movement/basic_avoidance
+	idle_behavior = /datum/idle_behavior/idle_random_walk/less_walking
+	planning_subtrees = list(
+		/datum/ai_planning_subtree/find_and_hunt_target/look_for_apcs,
+		/datum/ai_planning_subtree/simple_find_target,
+		/datum/ai_planning_subtree/basic_melee_attack_subtree,
+	)
+
+/mob/living/basic/festivus/attack_hand(mob/living/carbon/human/user, list/modifiers)
+	. = ..()
+	if(!user.combat_mode)
+		visible_message(span_warning("[src] crackles with static electricity!"))
+		for(var/obj/item/stock_parts/cell/cell in range(2, get_turf(src)))
+			cell.give(recharge_value)
+			cell.update_appearance()
+		for(var/mob/living/silicon/robot/robot in range(2, get_turf(src)))
+			if(robot.cell)
+				robot.cell.give(recharge_value)
+		for(var/obj/machinery/power/apc/apc_target in range(2, get_turf(src)))
+			if(apc_target.cell)
+				apc_target.cell.give(recharge_value)
+
+/datum/ai_planning_subtree/find_and_hunt_target/look_for_apcs
+	hunting_behavior = /datum/ai_behavior/hunt_target/apcs
+	hunt_targets = list(/obj/machinery/power/apc)
+	hunt_range = 6
+
+
+/datum/ai_planning_subtree/find_and_hunt_target/look_for_apcs
+	target_key = BB_LOW_PRIORITY_HUNTING_TARGET
+	finding_behavior = /datum/ai_behavior/find_hunt_target/apcs
+	hunting_behavior = /datum/ai_behavior/hunt_target/apcs
+	hunt_targets = list(/obj/machinery/power/apc)
+	hunt_range = 6
+
+/datum/ai_behavior/hunt_target/apcs
+	hunt_cooldown = 15 SECONDS
+	always_reset_target = TRUE
+	///how much charge we will give off to the APC we find
+	var/given_charge = 80
+
+/datum/ai_behavior/hunt_target/apcs/target_caught(mob/living/hunter, obj/machinery/power/apc/hunted)
+	new /obj/effect/particle_effect/sparks(hunted.loc)
+	hunted.cell.give(given_charge)
```

Comment:
Can you move this to the festivus' behavior itself? Aka, a 15 second cooldown on-attack ability that charges apcs? AI should be instructing festivus poles on how to act, not giving them unique functionality players can't do. If that makes sense